src/bijectors/ordered.jl

```diff
@@ -0,0 +1,60 @@
+"""
+    OrderedBijector()
+
+A bijector mapping ordered vectors in ℝᵈ to an unordered vectors in ℝᵈ.
+
+## See also
+- [Stan's documentation](https://mc-stan.org/docs/2_27/reference-manual/ordered-vector.html)
+  - Note that this transformation and its inverse are the _opposite_ of in this reference.
+"""
+struct OrderedBijector <: Bijector{1} end
+
+"""
+    ordered(d::Distribution)
+
+Returns a `Distribution` whose domain is now ordered vectors.
+"""
+ordered(d::Distribution) = Bijectors.transformed(d, OrderedBijector())
+
+(b::OrderedBijector)(y::AbstractVecOrMat) = _transform_ordered(y)
+
+function _transform_ordered(y::AbstractVector)
+    x = similar(y)
+    x[1] = y[1]
+    @inbounds for i = 2:size(x, 1)
+        x[i] = x[i - 1] + exp(y[i])
+    end
+
+    return x
+end
+
+function _transform_ordered(y::AbstractMatrix)
+    x = similar(y)
+    x[1, :] = y[1, :]
+    @inbounds for i = 2:size(x, 1)
+        x[i, :] = x[i - 1, :] + exp.(y[i, :])
+    end
```

Comment:
I guess it would be more efficient to operate on columns instead of rows but I assume it is required by the design of the matrix inputs? Also it seems quite allocation heavy.

Reply:
> I guess it would be more efficient to operate on columns instead of rows but I assume it is required by the design of the matrix inputs? 

Yeah, it's required by the inputs :confused:

> Also it seems quite allocation heavy.

Want me to add some `view`? I didn't bother because I figured: `exp.(...)` will already allocation.

Reply:
Btw, do you think it would be faster to do

```julia
@inbounds for j = 1:size(x, 2), i = 2:size(x, 1)
    x[i, j] = x[i - 1, j] + exp.(y[i, j])
end
```

?

Reply:
> I didn't bother because I figured: exp.(...) will already allocation.

`x[i - 1, :] + exp.(y[i, :])` will allocate an additional array, it seemed strange that one does not broadcast the whole expression and updates `x` in-place.

> Btw, do you think it would be faster to do?

I am not sure, best would be to benchmark it :slightly_smiling_face:

Reply:
> Btw, do you think it would be faster to do

Some quick checks tells me that yes this is the case. I'll just go with that then :+1: Also, guaranteed to have waaay fewer allocaitons (i.e. 1).